scripts/meta_analysis.py

```diff
@@ -553,12 +566,27 @@ def get_next_variant( studies : List[Study]) -> List[VariantData]:
             res.append(None)
             continue
 
-        for v in dats[i]:
-            if v.equalize_to(first):
+        # Flag tracks that only one variant (best match) is returned per study, if multiple variants equal to first
+        added=False
+        for j,v in reversed(list(enumerate(dats[i]))):
+            if v == first:
                 res.append(v)
-            else:
+                added=True
+                del dats[i][j]
+                s.put_back(dats[i])
+                break
+            if not v.is_equal(first):
                 s.put_back([v])
-        if len(res)<i+1:
+                del dats[i][j]
+        if not added:
+            for j,v in reversed(list(enumerate(dats[i]))):
```

Comment:
looks great! why are the same position variants reversed?

Reply:
Deleting elements from the list while iterating through is more safer/simpler this way (`del dats[i][j]`).